test/src/controllers/personsNameController.test.ts

```diff
@@ -31,11 +25,18 @@ describe("POST" + PERSONS_NAME, () => {
             });
         expect(res.status).toBe(302);
         expect(mocks.mockSessionMiddleware).toHaveBeenCalled();
+        expect(mocks.mockAuthenticationMiddleware).toHaveBeenCalled();
+        expect(res.header.location).toBe(BASE_URL + PERSONAL_CODE + "?lang=en");
     });
 
     // Test for incorrect form details entered, will return 400.
     it("should return status 400 after incorrect data entered", async () => {
-        const res = await router.post(BASE_URL + PERSONS_NAME);
+        const sendData = {
+            "first-name": "",
+            "middle-names": "",
+            "last-name": ""
+        };
+        const res = await router.post(BASE_URL + PERSONS_NAME).send(sendData); ;
         expect(res.status).toBe(400);
```

Comment:
Can you add expect for error message and further tests for other error scenarios

Reply:
Done